### DISCUSSION TITLE
calculating integral of stepped curve

### DISCUSSION THREAD

**Discussion by martin3000** [influxdata/influxdb#24315] [Q&A]
I have a sensor that delivers 60W for 55 minutes and then 5W for 5 minutes, and again 60W.
So the integral over 1h is 60*55*60+5*5*60=199500Ws . This is a mean value of around 55W.
But using influxQL 1.8
SELECT integral("value") FROM "sensor.influxtestsensor" GROUP BY time(1h)
I get 117000, which is a mean value of 32.5

How can I get the integral for the step function?

**Comment by thkoinje** [comment]
Here is another discussion with more details: #24437 
I guess it is the same topic.
